src/main/java/org/sonarlint/intellij/vcs/DefaultVcsService.kt

```diff
@@ -58,14 +58,20 @@ class DefaultVcsService @NonInjectable constructor(private val project: Project,
         val repositoriesEPs = ModuleVcsRepoProvider.EP_NAME.extensionList
         val repositories = repositoriesEPs.mapNotNull { it.getRepoFor(module, logger) }.toList()
         if (repositories.isEmpty()) {
+            logger.warn("No VCS repository found for module $module")
             return null
         }
         if (repositories.size > 1) {
             logger.warn("Several candidate Vcs repositories detected for module $module, cannot choose one")
             return null
         }
         val repo = repositories.first()
-        return repo.electBestMatchingServerBranchForCurrentHead(validConnectedEngine.getServerBranches(projectKey))
+        val resolvedBranch =
+            repo.electBestMatchingServerBranchForCurrentHead(validConnectedEngine.getServerBranches(projectKey))
+        if (resolvedBranch == null) {
```

Comment:
I guess this change is not strictly related to the change of IJ api

Reply:
No it was added for debug purposes when a test was failing. I will remove it